Guide/routing.markdown

```diff
@@ -0,0 +1,190 @@
+```
+
+Afterwards enable the routes for `PostsController` in `Web/FrontController.hs` like this:
+
+```haskell
+instance FrontController WebApplication where
+    controllers = 
+        [ -- ...
+        , parseRoute @PostsController
+        ]
+```
+
+Now you can open e.g. `/Posts` to access the `PostsAction`.
+
+## Url Generation
+
+Use `pathTo` to generate a path to a given action:
+
+```haskell
+pathTo ShowPostAction { postId = "adddfb12-da34-44ef-a743-797e54ce3786" }
+-- /ShowPost?postId=adddfb12-da34-44ef-a743-797e54ce3786
+```
+
+To generate a full url, use `urlTo`:
+
+```haskell
+urlTo NewUserAction
+-- http://localhost:8000/NewUser
+```
+
+## AutoRoute
+
+Let's say our `PostsController` is define in `Web/Types.hs` like this:
+
+```haskell
+data PostsController
+    = PostsAction
+    | NewPostAction
+    | ShowPostAction { postId :: !(Id Post) }
+    | CreatePostAction
+    | EditPostAction { postId :: !(Id Post) }
+    | UpdatePostAction { postId :: !(Id Post) }
+    | DeletePostAction { postId :: !(Id Post) }
+```
+
+Using `instance AutoRoute PostsController` will give us the following routing:
+
+```haskell
+GET /Posts                          => PostsAction
+GET /NewPost                        => NewPostAction
+GET /ShowPost?postId={postId}       => ShowPostAction { postId }
+POST /CreatePost                    => CreatePostAction
+GET /EditPost?postId={postId}       => EditPostAction { postId }
+POST /UpdatePost?postId={postId}    => UpdatePostAction { postId }
+PATCH /UpdatePost?postId={postId}   => UpdatePostAction { postId }
+DELETE /DeletePost?postId={postId}  => DeletePostAction { postId }
+```
+
+The urls are very close to the actual action which is called. Action parameters are taken automatically from the request query. This design helps you to always know which action is actually called, when requesting an url. 
+
+### AutoRoute & Beautiful Urls
+
+Lots of modern browser don't even show the full url bar anymore (e.g. Safari and most mobile browsers). Therefore AutoRoute doesn't aim to generate the "most" beautiful urls out of the box. It's rather optimized for the needs of developers. If you need beautiful urls for SEO reasons, instead of using AutoRoute you can use the more manual APIs of IHP Routing. 
+
+### Multiple Parameters
+
+An action constructor can have multiple parameters:
+
+```haskell
+data PostsController = EditPostAction { postId :: !(Id Post), userId :: !(Id Post) }
+```
+
+This will generate a routing like:
+
+```haskell
+GET /EditPost?postId={postId}&userId={userId} => EditPostAction { postId, userId }
+```
+
+### Parameter Types
+
+AutoRoute by default only works with UUID arguments (and Id types like `Id Post`). If you have a controller like `data HelloWorldController = HelloAction { name :: Text }` where you have text parameter, you have to configure AutoRoute like this:
+
+```haskell
+instance AutoRoute HelloWorldController where
+    parseArgument = parseTextArgument
```

Comment:
- Is there a `parseIntArgument`?
- How would one use own arguments there, in a type safe way? E.g. a type for a Youtube-style shortid.

Reply:
`parseIntArgument` has been added now :)

Also added a section on creating a custom `parseArgument`